fedmsg_meta_fedora_infrastructure/fasshim.py

```diff
@@ -90,6 +90,50 @@ def avatar_url_from_email(email, size=64, default='retro', dns=False):
         hash = md5(email.encode('utf-8')).hexdigest()
         return "https://seccdn.libravatar.org/avatar/%s?%s" % (hash, query)
 
+def make_fasjson_cache(**config):
+    global _fas_cache
+    if _fas_cache:
+        return _fas_cache
+    
+    log.warn("No previous fasjson cache found. Looking to rebuild.")
+
+    try:
+        import fasjson_client
+    except ImportError:
+        log.warn("No fasjson-client installed.  Not caching fasjson.")
+    
+    creds = config['fas_credentials']
+    
+    default_url = 'https://admin.fedoraproject.org/accounts/'
+    
+    try:
+        client = fasjson_client.Client(url=creds.get('base_url', default_url))
+    except fasjson_client.errors.ClientSetupError as e:
+        log.error("Something went wrong setting up fasjson client with error: %s" % e)
+    try:
+        response = client.list_users(page_size=1000)
+        _add_to_cache(response.result)
+        total_pages = response.page.get('total_pages')
+        for x in range(1, total_pages):
```

Comment:
You're requesting page 1 twice, because that's what you got before the for loop. Also, `range()` will stop at the int before the end boundary so you'll miss the last page.
You can probably structure that around a while loop to make things simpler (increment the page number on each step until it reaches `total_pages`). Or maybe simplicity is in the eye of the beholder here ;-)

Reply:
thanks, I thought I had set the for to start at `2`, obviously not. Yeah a while loop is simpler, will do.